### DISCUSSION TITLE
How I train Fooocus? (If I need to)

### DISCUSSION THREAD

**Discussion by obit8** [lllyasviel/Fooocus#1332] [Q&A] [answered]
Good morning, I am asking for help because I am a Midjourney user and I usually use that model mainly for portraits. I have noticed that on Fooocus I can't achieve faces similar to those I get on Midjourney. Specifically, I'm trying with the Faceswap function to achieve a slightly thinner face, but even when writing 'thin face' in the prompt, nothing changes.

So, my question is:

Is there a way to instruct Foocus, or is there a way to give more priority to certain parameters over others?

**Comment by FrietbakkerThjeu** [comment]
Lora's

**Reply by obit8** [reply]
What do you mean? I know lora's as a style...I should use it to teach to fooocus how to stylize my images? How?

**Reply by FrietbakkerThjeu** [reply]
use Loras to train Fooocus the types of faces you like. Thats how you get the consistent faces, the margot robbie Lora for instance. there are extensive tutorials available on youtube. you can find a whole lot of those on civitai

**Reply by obit8** [reply]
Ok i'm trying to find some tutorial becuase I don't even know where to start to train LORA, I mean that I don't even know wich tools to use.
thank you for the advice

**Reply by FrietbakkerThjeu** [reply]
You will need massive VRAM capacity to train loras offline. I use Google Colab to train. it basically comes down to this: You input a dataset of images, with corresponding .txt files where prompts about the image are defined. Lets say you want to train a distinct "smile": you define the word "smile" in the file. The training software - i believe most training is done with Kohya-ss - is teaching stable diffusion to recognise what image corresponds best with the word smile.

so when u want a person with a smile, it knows what type of smile you are looking for. They use this for certain types of clothing and art styles also.

**Reply by obit8** [reply]
Yes, I am reading various tutorials, it's not easy because each tutorial takes something for granted. For instance, some assume that one knows how to use Google Colab, but I didn't even know it existed.

I haven't yet found a really well-made tutorial that explains from the beginning, for someone who has never done any of this, what you need and how to use it... but I'll get there ;)

And then I will write a good tutorial for the next that will be in my same situation :D

**Reply by FrietbakkerThjeu** [reply]
If you need any help just ask, i was in your position a couple weeks ago. Google Colab is the way to go, i used it to train on Stable diffusion 1.5 base model. If you want to train in SDXL you will need (atleast) colab pro which isn't free. A lot of those tutorials are made by people who dont really know what all the settings are for, the best way to start is to follow this:
https://colab.research.google.com/drive/1-D0l9UdkmUx25EonusH0ZGtzqqPWgo_c

or this guide:
https://colab.research.google.com/github/Linaqruf/kohya-trainer/blob/main/kohya-LoRA-dreambooth.ipynb

Python will return a lot of frustrating nonsensical errors, its almost always about insufficient VRAM

**Reply by obit8** [reply]
I can't open these links, don't know why.

**Reply by FrietbakkerThjeu** [reply]
me neither, fixed it now

**Reply by portorro** [reply]
I do not need google colab or none of that i have a powerfull computer and python is my friend is there a training step by step tutorial without fancy 3rd party software things?

**Reply by codezeros** [reply]
Hi @FrietbakkerThjeu  i would also like to fine-tune and train my own LLMs is there any guidelines for that with either from any existing model like llama or any other model or from scratch and also for stable diffusion models with fully base model or as loras on locally using GPU?

**Reply by mashb1t** [reply, marked as answer]
perfectly answered by @IPv6 in https://github.com/lllyasviel/Fooocus/discussions/2928#discussioncomment-9455210

**Comment by portorro** [comment]
https://youtu.be/y2J7EZUk_a0?si=v_HcbtwlHD4NiYB4

Thats all ypu need. I TESTED IT. all the you cant do it without google is bs.
